typescript/infra/scripts/funding/vanguard.ts

```diff
@@ -0,0 +1,233 @@
+import chalk from 'chalk';
+import { BigNumber } from 'ethers';
+import { formatUnits, parseUnits } from 'ethers/lib/utils.js';
+import yargs from 'yargs';
+
+import { ChainMap } from '@hyperlane-xyz/sdk';
+import { Address, rootLogger } from '@hyperlane-xyz/utils';
+
+import { Contexts } from '../../config/contexts.js';
+import { Role } from '../../src/roles.js';
+import { getEnvironmentConfig } from '../core-utils.js';
+
+const VANGUARDS = [
+  'vanguard0', // regular
+  'vanguard1', // regular
+  'vanguard2', // regular
+  'vanguard3', // 5x gas cap
+  'vanguard4', // 5x gas cap
+  'vanguard5', // 10x gas cap
+] as const;
+
+type VanguardName = (typeof VANGUARDS)[number];
+type VanguardBalance = {
+  chain: string;
+  vanguard: VanguardName;
+  balance: string;
+};
+
+const TOKEN_DECIMALS = 18;
+const MIN_FUNDING_AMOUNT = parseUnits('0.05', TOKEN_DECIMALS);
+
+const VANGUARD_ENVIRONMENT = 'mainnet3';
+const VANGUARD_ADDRESSES: Record<VanguardName, Address> = {
+  vanguard0: '0xbe2e6b1ce045422a08a3662fffa3fc5f114efc3d',
+  vanguard1: '0xdbcd22e5223f5d0040398e66dbb525308f27c655',
+  vanguard2: '0x226b721316ea44aad50a10f4cc67fc30658ab4a9',
+  vanguard3: '0xcdd728647ecd9d75413c9b780de303b1d1eb12a5',
+  vanguard4: '0x5401627b69f317da9adf3d6e1e1214724ce49032',
+  vanguard5: '0x6fd953d1cbdf3a79663b4238898147a6cf36d459',
+};
+
+const VANGUARD_NETWORKS = [
+  'base',
+  'arbitrum',
+  'optimism',
+  'ethereum',
+  'bsc',
+] as const;
+
+const VANGUARD_FUNDING_CONFIGS: Record<
+  (typeof VANGUARD_NETWORKS)[number],
+  Record<VanguardName, string>
+> = {
+  base: {
+    vanguard0: '1', // 1x gas cap
+    vanguard1: '1', // 1x gas cap
+    vanguard2: '1', // 1x gas cap
+    vanguard3: '1', // 5x gas cap
+    vanguard4: '1', // 5x gas cap
+    vanguard5: '1', // 10x gas cap
```

Comment:
the comments here are just so we are aware that e.g. vanguard 5 is configured with a 10 higher gas cap but this doesn't directly impact the funding right? it's just for context?

Reply:
i'll remove the comments from the config and leave it just at the `VANGUARDS` const up top to reduce confusion